### ISSUE TITLE
Intermittent errors on clean task

### ISSUE THREAD

**Issue by yinzara** [NMica/NMica#4]
We experience intermittent "Directory is not empty" errors with this plugin during the clean task.

This is related to the issue indicated in:
http://stackoverflow.com/questions/329355/cannot-delete-directory-with-directory-deletepath-true

And the use of Directory.Delete(path, true);

**Comment by macsux** [comment]
I merged in the request, but something got changed with how MSBuild processes custom tasks and the existing way this lib was packaged no longer works. MSBuild is a giant pile of burning garbage due to having two separate versions: one that comes with Visual Studio written in .NET Framework and one that comes with Core SDK that uses .NET Core. This creates a problem creating custom tasks as they need to be implemented twice, which is what causes 90% of the complexity of this project. There were hacks employed to allow me to target both msbuilds and use a nuget package inside my custom task using approach here https://github.com/AArnott/Nerdbank.MSBuildExtension, but it seems some recent changes broke this approach. Need to figure out a workaround. 

If you want to assist I could use an extra help with this cuz it's driving me mental atm. The ContextIsolatedTask works fine for Visual Studio version of msbuild, but is broken for .NET Core now. It's even more bizarre that this stopped working AFTER I've pulled latest of docker images of the official .NET SDK I already had (patch versions got incremented).
